tests/baselines/reference/optionalPropertyAssignableToStringIndexSignature.errors.txt

```diff
@@ -0,0 +1,47 @@
+tests/cases/conformance/types/typeRelationships/assignmentCompatibility/optionalPropertyAssignableToStringIndexSignature.ts(6,1): error TS2322: Type '{ k1: string | undefined; }' is not assignable to type '{ [key: string]: string; }'.
+  Property 'k1' is incompatible with index signature.
+    Type 'string | undefined' is not assignable to type 'string'.
+      Type 'undefined' is not assignable to type 'string'.
+tests/cases/conformance/types/typeRelationships/assignmentCompatibility/optionalPropertyAssignableToStringIndexSignature.ts(10,1): error TS2322: Type '{ 1?: string | undefined; }' is not assignable to type '{ [key: number]: string; }'.
+  Property '1' is incompatible with index signature.
+    Type 'string | undefined' is not assignable to type 'string'.
+      Type 'undefined' is not assignable to type 'string'.
+tests/cases/conformance/types/typeRelationships/assignmentCompatibility/optionalPropertyAssignableToStringIndexSignature.ts(13,5): error TS2322: Type '{ k1?: undefined; }' is not assignable to type '{ [key: string]: string; }'.
+  Property 'k1' is incompatible with index signature.
+    Type 'undefined' is not assignable to type 'string'.
+
+
+==== tests/cases/conformance/types/typeRelationships/assignmentCompatibility/optionalPropertyAssignableToStringIndexSignature.ts (3 errors) ====
+    declare let optionalProperties: { k1?: string };
+    declare let undefinedProperties: { k1: string | undefined };
+    
+    declare let stringDictionary: { [key: string]: string };
+    stringDictionary = optionalProperties;  // ok
+    stringDictionary = undefinedProperties; // error
+    ~~~~~~~~~~~~~~~~
+!!! error TS2322: Type '{ k1: string | undefined; }' is not assignable to type '{ [key: string]: string; }'.
+!!! error TS2322:   Property 'k1' is incompatible with index signature.
+!!! error TS2322:     Type 'string | undefined' is not assignable to type 'string'.
+!!! error TS2322:       Type 'undefined' is not assignable to type 'string'.
+    
+    declare let probablyArray: { [key: number]: string };
+    declare let numberLiteralKeys: { 1?: string };
+    probablyArray = numberLiteralKeys;  // error
+    ~~~~~~~~~~~~~
+!!! error TS2322: Type '{ 1?: string | undefined; }' is not assignable to type '{ [key: number]: string; }'.
+!!! error TS2322:   Property '1' is incompatible with index signature.
+!!! error TS2322:     Type 'string | undefined' is not assignable to type 'string'.
+!!! error TS2322:       Type 'undefined' is not assignable to type 'string'.
+    
+    declare let optionalUndefined: { k1?: undefined };
+    let dict: { [key: string]: string } = optionalUndefined; // error
+        ~~~~
+!!! error TS2322: Type '{ k1?: undefined; }' is not assignable to type '{ [key: string]: string; }'.
+!!! error TS2322:   Property 'k1' is incompatible with index signature.
+!!! error TS2322:     Type 'undefined' is not assignable to type 'string'.

```

Comment:
This is the only mildly interesting thing about this PR. We check if the property type is `undefined` before filtering `undefined` out of it in order to keep this error.

Reply:
I actually think the `?: string | undefined` -> `[x: string]: string` one is actually weirder, but it is consistent.